openedx/core/djangoapps/catalog/utils.py

```diff
@@ -49,14 +49,31 @@ def get_programs(uuid=None):
 
     uuids = cache.get(PROGRAM_UUIDS_CACHE_KEY, [])
     if not uuids:
-        logger.warning('Program UUIDs are not cached.')
+        logger.warning('Failed to get program UUIDs from the cache.')
 
     programs = cache.get_many([PROGRAM_CACHE_KEY_TPL.format(uuid=uuid) for uuid in uuids])
     programs = list(programs.values())
 
+    # The get_many above sometimes fails to bring back details cached on one or
+    # more Memcached nodes. It doesn't look like these keys are being evicted.
+    # 99% of the time all keys come back, but 1% of the time all the keys stored
+    # on one or more nodes are missing from the result of the get_many. One
+    # get_many may fail to bring these keys back, but a get_many occurring
+    # immediately afterwards will succeed in bringing back all the keys. This
+    # behavior can be mitigated by trying again for the missing keys, which is
+    # what we do here. Splitting the get_many into smaller chunks may also help.
```

Comment:
Should we craft a ticket to look into the root cause of this?  Does this affect all `get_many()` calls, or just ones involving programs?